### PR TITLE
WT-8936 Update CMake files to be able to use the Memkind library

### PR DESCRIPTION
**Pull request by etienneptl**
The changes have been tested by @fedorova and worked as expected.

### CODE REVIEW

**Review by korteland — APPROVED** [review]
LGTM!

File: cmake/configs/base.cmake [hunk 1]
```diff
@@ -140,6 +140,16 @@ config_bool(
     DEPENDS_ERROR ON "Failed to find lz4 library"
 )
 
+config_bool(
+    ENABLE_MEMKIND
+    "Use MEMKIND as the backend allocator"
```
Comment (etienneptl) [inline cmake/configs/base.cmake:145]: @fedorova, what would be a better description?

Comment (fedorova) [reply cmake/configs/base.cmake:145]: A better description would be: "Enable the memkind library, needed for NVRAM or SSD block caches."

**Review by fedorova — APPROVED** [review]
LGTM

File: cmake/configs/base.cmake (hunk 1 above)
Comment (etienneptl) [reply cmake/configs/base.cmake]: Thanks!

**Pull request merged**